src/common/stats/StatsManager.cpp

```diff
@@ -81,28 +73,27 @@ int32_t StatsManager::registerHisto(folly::StringPiece counterName,
 
     // Insert the Histogram
     sm.histograms_.emplace_back(
-        std::make_pair(
-            std::make_unique<std::mutex>(),
-            std::make_unique<HistogramType>(
-                bucketSize,
-                min,
-                max,
-                StatsType(60, {seconds(5), seconds(60), seconds(600), seconds(3600)}))));
-    int32_t index = - sm.histograms_.size();
+        std::make_pair(std::make_unique<std::mutex>(),
+                       std::make_unique<HistogramType>(
+                           bucketSize,
+                           min,
+                           max,
+                           StatsType(60, {seconds(5), seconds(60), seconds(600), seconds(3600)}))));
+    int32_t index = -sm.histograms_.size();
     sm.nameMap_[name] = index;
 
-    LOG(INFO) << "registerHisto, bucketSize: " << bucketSize
-              << ", min: " << min << ", max: " << max;
+    LOG(INFO) << "registerHisto, bucketSize: " << bucketSize << ", min: " << min
+              << ", max: " << max;
     return index;
 }
 
-
 // static
 void StatsManager::addValue(int32_t index, VT value) {
     using std::chrono::seconds;
     CHECK_NE(index, 0);
 
     auto& sm = get();
+    folly::RWSpinLock::ReadHolder rh(sm.nameMapLock_);
```

Comment:
Do we really need the lock?  All metrics have been registered when starting. 
It hurt the performance.

Reply:
In fact, it's registered when running.

Reply:
Really,  why register them when running.  It is not a good way for performance. At least on storage-side,  it is required at beginning.

Reply:
Emm, as  @sherman-the-tank say there will be some stats need registered when running, so we can's assume it's ready when initialization.